src/app/state/actions/index.tsx

```diff
@@ -405,11 +405,15 @@ export const handleProviderCallback = (provider: AuthenticationProvider, paramet
                         provider: provider,
                     }
                 }
-            )
+            );
         }
-        const nextPage = persistence.load(KEY.AFTER_AUTH_PATH);
-        persistence.remove(KEY.AFTER_AUTH_PATH);
-        history.push(nextPage?.replace("#!", "") || "/account");
+
+        // On first login (registration), redirect to /account if there is no after-auth path.
+        // After-auth path should take presedence for the case where users register while following a group invite - /account?authToken=GROUP1, for example.
+        // They will see the required account information modal either way on registration.
+        const nextPage = persistence.pop(KEY.AFTER_AUTH_PATH)?.replace("#!", "");
```

Comment:
```suggestion
        const nextPage = persistence.pop(KEY.AFTER_AUTH_PATH);
```
Tracing the history of the weird `replace("#!", "")`, it seems to be a throwback to the [very very old AngularJS world](https://github.com/isaacphysics/isaac-app/commit/3df8034065a185777a9a3e17926443987b91a0a6#diff-62b9ec54ab5471ceea09f76d449f259528cd43c9261ccaf6f9d5ae61a4a38148R19)! I am sure we don't need it any more.